aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -37,8 +38,11 @@ module aptos_token_objects::token {
     struct Token has key {
         /// The collection from which this token resides.
         collection: Object<Collection>,
+        /// Deprecated in favor of index inside TokenAppendix1.
+        /// Will be populated until concurrent_token_v2_enabled feature flag is enabled.
+        ///
         /// Unique identifier within the collection, optional, 0 means unassigned
-        index: u64,
+        deprecated_index: u64,
```

Comment:
This would violate the backward compat policy for module ugprades :)

Reply:
haha, I wanted to ask that.

Would it, as it is just name, and BCS is ignoring it. are names themselves enforced not to change by module upgrade?

otherwise I can just add a comment to mark it that way

Reply:
does it? I thought field name is irrelevant

Reply:
It doesn't matter for function args, it does for struct fields! Just tested it

Reply:
if it does, it sounds like unnecessary constraint that we may be able to relax? let's discuss in #move-eng or somewhere

Reply:
While it won't break anything for BCS compatibility, it will break it for JSON: https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/python/sdk/aptos_sdk/aptos_token_client.py#L118

might be too risky a move...